controllers/storageclassrequest/storageclassrequest_controller_test.go

```diff
@@ -218,6 +267,201 @@ func TestProfileReconcile(t *testing.T) {
 	assert.NoError(t, err, caseLabel)
 }
 
+func TestStorageProfileCephBlockPool(t *testing.T) {
+	var err error
+	var caseCounter int
+
+	var primaryTestCases = []struct {
+		label            string
+		expectedPoolName string
+		failureExpected  bool
+		createObjects    []runtime.Object
+		storageProfile   *v1.StorageProfile
+	}{
+		{
+			label:            "valid profile",
+			expectedPoolName: "test-valid-blockpool",
+			failureExpected:  false,
+			storageProfile:   validStorageProfile,
+			createObjects: []runtime.Object{
+				&rookCephv1.CephBlockPool{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-valid-blockpool",
+						Namespace: namespaceName,
+						Labels: map[string]string{
+							controllers.StorageConsumerNameLabel: fakeStorageConsumer.Name,
+							controllers.StorageProfileSpecLabel:  validStorageProfile.GetSpecHash(),
+						},
+					}, Spec: rookCephv1.NamedBlockPoolSpec{
+						Name: "spec",
+						PoolSpec: rookCephv1.PoolSpec{
+							FailureDomain: "zone",
+							DeviceClass:   deviceClass,
+							Parameters:    map[string]string{},
+						},
+					},
+				},
+			},
+		},
+		{
+			label:            "rejected profile",
+			expectedPoolName: "test-rejected-blockpool",
+			failureExpected:  true,
+			storageProfile:   rejectedStorageProfile,
+			createObjects: []runtime.Object{
+				&rookCephv1.CephBlockPool{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-rejected-blockpool",
+						Namespace: namespaceName,
+						Labels: map[string]string{
+							controllers.StorageConsumerNameLabel: fakeStorageConsumer.Name,
+							controllers.StorageProfileSpecLabel:  rejectedStorageProfile.GetSpecHash(),
+						},
+					}, Spec: rookCephv1.NamedBlockPoolSpec{
+						Name: "spec",
+						PoolSpec: rookCephv1.PoolSpec{
+							FailureDomain: "zone",
+							DeviceClass:   deviceClass,
+							Parameters:    map[string]string{},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, c := range primaryTestCases {
+		caseCounter++
+		caseLabel := fmt.Sprintf("Case %d: %s", caseCounter, c.label)
+		fmt.Println(caseLabel)
+
+		r := createFakeReconciler(t)
+		if strings.Contains(c.label, "rejected") {
+			r.storageCluster.Spec.DefaultStorageProfile = rejectedStorageProfile.Name
+		}
+		r.StorageClassRequest.Spec.Type = "blockpool"
+
+		r.StorageClassRequest.Spec.StorageProfile = c.storageProfile.Name
+
+		c.createObjects = append(c.createObjects, c.storageProfile)
+		c.createObjects = append(c.createObjects, fakeStorageConsumer)
+
+		fakeClient := fake.NewClientBuilder().WithScheme(r.Scheme).WithRuntimeObjects(c.createObjects...)
+		r.Client = fakeClient.Build()
+
+		_, err = r.reconcilePhases()
+		if c.failureExpected {
+			assert.Error(t, err, caseLabel)
+			continue
+		}
+		assert.NoError(t, err, caseLabel)
+
+		assert.Equal(t, c.expectedPoolName, r.cephBlockPool.Name, caseLabel)
+
+		if strings.Contains(c.expectedPoolName, "valid") {
```

Comment:
we can check for the exact pool name, right? Any specific reason for contains

Reply:
Thought performance and readability wise it's better for cases where there's a unique sub-string

Reply:
I imagine the strategy is to consistently name the pools on whether they are expected to be valid or invalid/rejected per test case, and just use that as the check. Still, given that you're checking the Phase for the rejected test case below, it could make sense to also verify the Phase in the "valid" cases as well. `if c.expectedPhase == PhaseRejected` as the conditional, for example, then verify the actual values.